packages/create-amplify/templates/basic-auth-data/amplify/data/resource.ts

```diff
@@ -1,12 +1,58 @@
 import { type ClientSchema, a, defineData } from '@aws-amplify/backend';
 
+/*== STEP 1 ===============================================================
+The section below creates a Todo database table with a "content" field. Try
+adding a new "isDone" field as a boolean. The authorization rules below
+specify that owners, authenticated via your Auth resource can "create",
+"read", "update", and "delete" their own records. Public users,
+authenticated via an API key, can only "read" records.
+=========================================================================*/
 const schema = a.schema({
-  Todo: a.model({
-    name: a.string(),
-    description: a.string(),
-  }),
+  Todo: a
+    .model({
+      content: a.string(),
+    })
+    .authorization([a.allow.owner(), a.allow.public().to(['read'])]),
 });
 
 export type Schema = ClientSchema<typeof schema>;
 
-export const data = defineData({ schema });
+export const data = defineData({
+  schema,
+  authorizationModes: {
+    defaultAuthorizationMode: 'API_KEY',
+    // API Key is used for a.allow.public() rules
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+
+/*== STEP 2 ===============================================================
+Go to your frontend source code. From your client-side code, generate a
+Data client to make CRUDL requests to your table. (THIS SNIPPET WILL ONLY
+WORK IN THE FRONTEND CODE FILE.)
+
+Using JavaScript or Next.js React Server Components, Middleware, Server 
+Actions or Pages Router? Review how to generate Data clients for those use
+cases: https://docs.amplify.aws/gen2/build-a-backend/data/connect-to-API/
+=========================================================================*/
```

Comment:
what about mobile devs?

Reply:
The data-schema client only supports JS-based mobile platforms currently (React Native/Expo)

Reply:
`data-schema` yes, but `data/resource.ts` not necessarily. so to give better instructions this might be expanded to say something about mobile use cases.

I just hooked up `data/resource.ts` backend to android app while testing this change https://github.com/aws-amplify/amplify-backend/pull/641

Reply:
mobile messaging will be handled in a completely separate approach. Let's not block this PR on that for now. Please discuss with Nikhil for the final mobile messaging.